### PR TITLE
RoleRequirement - Include stderr in the error message

### LINKED ISSUE #41336: ansible-galaxy hides scm errors
##### SUMMARY

When downloading roles from SCM repos using `ansible-galaxy -r requirements.yml`, the output from the SCM commands is dumped to `/dev/null`. This makes it very difficult to debug issues, particularly in non-interactive environments where all you have is a log to go by. It would be better to capture the SCM command output and display it (perhaps only when ANSIBLE_VERBOSITY > 1 to minimize noise).

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

ansible-galaxy
##### ANSIBLE VERSION

```
 ansible-galaxy 2.5.0
config file = None
configured module search path = ['/var/lib/jenkins/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
ansible python module location = /opt/ansible/2.5.0.0/lib/python3.5/site-packages/ansible
executable location = /opt/ansible/2.5.0.0/bin/ansible-galaxy
python version = 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

`requirements.yml`
```yaml
- role: myrole
  src: <scm url>
  scm: git
```

It is assumed that SCM url is either inaccessible or otherwise incorrect in a way that ansible-galaxy cannot detect

`ansible-galaxy install -r requirements.yml`

##### EXPECTED RESULTS

`
[WARNING]: - myrole was NOT installed successfully: - command git clone <scm_url> myrole failed in directory /some/tmp/dir (rc=128). Command output follows:
<command output>
`
##### ACTUAL RESULTS

```
[WARNING]: - myrole was NOT installed successfully: - command git clone
14:38:15     baseline/cis-level-1: <scm url> myrole failed in directory /tmp/tmpglah_hf5 (rc=128)
```

Or something like that.

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #41336

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/playbook/role/requirement.py

### CODE REVIEW

File: lib/ansible/playbook/role/requirement.py [hunk 1]
```diff
@@ -145,11 +145,11 @@ def run_scm_cmd(cmd, tempdir):
             except Exception as e:
                 ran = " ".join(cmd)
                 display.debug("ran %s:" % ran)
-                display.debug("\tstdout: " + stdout)
-                display.debug("\tstderr: " + stderr)
+                display.debug("\tstdout: " + to_native(stdout))
```
Comment (samdoran) [inline lib/ansible/playbook/role/requirement.py]:
```suggestion
                display.debug("\tstdout: " + to_text(stdout))
```

File: lib/ansible/playbook/role/requirement.py [hunk 2]
```diff
@@ -145,11 +145,11 @@ def run_scm_cmd(cmd, tempdir):
...
+                display.debug("\tstderr: " + to_native(stderr))
```
Comment (samdoran) [inline lib/ansible/playbook/role/requirement.py]:
```suggestion
                display.debug("\tstderr: " + to_text(stderr))
```

**Comment by samdoran** [comment]
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Review by jamescassell — APPROVED** [review]
shipit

**Pull request merged**